test/e2e/framework/pod/resize.go

```diff
@@ -330,28 +330,31 @@ func VerifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework
 			} else {
 				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
 			}
-			cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
+
+			expectedCPULimits := GetCPULimitCgroupExpectations(cpuLimit.MilliValue())
 			if cpuLimit.IsZero() {
-				cpuQuota = -1
+				if *podOnCgroupv2Node {
+					expectedCPULimits = []string{fmt.Sprintf("%s %s", "max", CPUPeriod)}
+				} else {
+					str := strconv.FormatInt(-1, 10)
+					expectedCPULimits = []string{fmt.Sprintf("%s %s", str, CPUPeriod)}
+				}
 			}
```

Comment:
I think this logic should be applied in the pod-level resources case too? cc @ndixita 

If that's the case, move it into the `GetCPULimitCgroupExpectations` function.

Reply:
moved this into `GetCPULimitCgroupExpectations`